nucypher/cli/commands/contacts.py

```diff
@@ -0,0 +1,139 @@
+type_option = click.option('--type', 'character_flag', help="Type of card: (A)lice or (B)ob.", type=click.STRING, required=False)
+verifying_key_option = click.option('--verifying-key', help="Alice's or Bob's verifying key as hex", type=click.STRING, required=False)
+encrypting_key_option = click.option('--encrypting-key', help="An encrypting key as hex", type=click.STRING, required=False)
+nickname_option = click.option('--nickname', help="Human-readable nickname / alias for a card", type=click.STRING, required=False)
+
+
+@click.group()
+def contacts():
+    """Lightweight contacts utility to store public keys of known ("character cards") Alices and Bobs."""
+
+
+@contacts.command()
+@click.argument('query')
+@click.option('--qrcode', help="Display the QR code representing a card to the console", is_flag=True, default=None)
+def show(query, qrcode):
+    """View existing character card"""
+    emitter = StdoutEmitter()
+    try:
+        card = select_card(emitter=emitter, card_identifier=query)
+    except Card.UnknownCard as e:
+        return emitter.error(str(e))
+    paint_single_card(emitter=emitter, card=card, qrcode=qrcode)
+
+
+@contacts.command('list')
+def _list():
+    """Show all character cards"""
+    emitter = StdoutEmitter()
+    card_directory = Card.CARD_DIR
+    try:
+        card_filepaths = os.listdir(card_directory)
+    except FileNotFoundError:
+        os.mkdir(Card.CARD_DIR)
+        card_filepaths = os.listdir(card_directory)
+    if not card_filepaths:
+        emitter.error(f'No cards found at {card_directory}.  '
+                      f"To create one run 'nucypher {contacts.name} {create.name}'.")
+    cards = list()
+    for filename in card_filepaths:
+        card_id, ext = filename.split('.')
+        card = Card.load(identifier=card_id)
+        cards.append(card)
+    paint_cards(emitter=emitter, cards=cards, as_table=True)
+
+
+@contacts.command()
+@type_option
+@encrypting_key_option
+@verifying_key_option
+@nickname_option
+@option_force
+def create(character_flag, verifying_key, encrypting_key, nickname, force):
+    """Store a new character card"""
+    emitter = StdoutEmitter()
+
+    if not all((character_flag, verifying_key, encrypting_key)) and force:
+        emitter.error(f'--verifying-key, --encrypting-key, and --type are required with --force enabled.')
+
+    if not character_flag:
+        from constant_sorrow.constants import ALICE, BOB
+        choice = click.prompt('Enter Card Type - (A)lice or (B)ob', type=click.Choice(['a', 'b'], case_sensitive=False))
+        flags = {'a': ALICE, 'b': BOB}
+        character_flag = flags[choice]
+
+    if not verifying_key:
+        verifying_key = click.prompt('Enter Verifying Key', type=UMBRAL_PUBLIC_KEY_HEX)
+    verifying_key = bytes.fromhex(verifying_key)  # TODO: Move / Validate
+
+    if character_flag is BOB:
+        if not encrypting_key:
+            encrypting_key = click.prompt('Enter Encrypting Key', type=UMBRAL_PUBLIC_KEY_HEX)
```

Comment:
How come we only prompt for encrypting key when the character flag is 'BOB' - what about alice?

I suspect this is cause of the 'None' in the below.
```bash
 [10] → nucypher contacts create
Enter Card Type - (A)lice or (B)ob (a, b): a
Enter Verifying Key: 027b8c9cf91929e02446709b9d96faca6f6e6377f52eed2f9c14e08ac1db9267e4
Traceback (most recent call last):
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/bin/nucypher", line 33, in <module>
    sys.exit(load_entry_point('nucypher', 'console_scripts', 'nucypher')())
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/derek/.local/share/virtualenvs/nucypher-jOttX5sw/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/derek/Documents/Github/repos/forks/kieran/nucypher/nucypher/cli/commands/contacts.py", line 109, in create
    card_id_hex = new_card.id.hex()
  File "/Users/derek/Documents/Github/repos/forks/kieran/nucypher/nucypher/policy/identity.py", line 250, in id
    return self.__hash(self.__payload)
  File "/Users/derek/Documents/Github/repos/forks/kieran/nucypher/nucypher/policy/identity.py", line 152, in __payload
    payload = b''.join(bytes(e) for e in elements)
  File "/Users/derek/Documents/Github/repos/forks/kieran/nucypher/nucypher/policy/identity.py", line 152, in <genexpr>
    payload = b''.join(bytes(e) for e in elements)
TypeError: cannot convert 'NoneType' object to bytes
```

Reply:
we may need to add some CLI tests.

Reply:
> How come we only prompt for encrypting key when the character flag is 'BOB' - what about alice?

There is no such key for alice, or at least, only alice verifying key is required for the grant and  retrieve lifecycle.

Reply:
Ah ok - but it doesn't seem that the Card handles a lack of encrypting key well, a la `NoneType`

Reply:
:heavy_check_mark: